README.md

```diff
@@ -15,26 +15,75 @@ Once you have implemented the `sa` program (and tested it to the best of your ab
+by running code from project 1. Running this on a Windows mashine (priviously only run on Linux and Mac) 
+uncovered two problems in the code. 
+1. Since Windows saves line breaks as \r\n a line break in files read with "r" will be counted as two chars,
+which will make the string length longer than expected. This was solved by reading the file in binary mode "rb". 
+2. When making the strict border array, the code was missing a check for whether bax[i]==0. Without this, the code tried 
+to find index -1 of bax. By coincidence index -1 pointed to a place in memory which contained 0, the desired value, 
+in every test. Probably since there was some relation between the location of bax and some other value. 
+In Linux that is. When run on Windows, index -1 gave a very small negative number. This is not a problem with Windows, 
+but an implementation error only discovered, because we ran it on Windows. 
 
+The problems show the importance of running code on different OS, both to test whether it can run and to uncover errors
+which were hidden by sheer coincidence. 
 
+The week after this meeting each group member focused on a different project. Anna focused on project 1, 
+Jonathan on project 2, and Gaia on project 3. Lastly, we used one full day together with project 3's code. 
 
-## Insights you may have had while implementing the algorithm
+We were close to not finish the code before deadline. Unfortunately we did not finish the report within this time. 
+One problem we had was timing of the algorithms. We had problems with the editor being struck, since we made a lot of files.
+A lot of time was wasted, waiting for the editor to index the files, only to crash. When we changed our strategy another 
+problem became apperent. Since we did not have the time or knowledge to make proper use of free() we would quickly run 
+out of memory when trying to run our tests, since we wanted to collect data from different n and m. We ran smaller 
+batches to compensate. Ideally, we would have fixed it with better memory control. 
+ 
+
+
+Regarding the implementation itself, the problems were mostly with the understanding of C and of-by-one errors. 
 
-## Problems encountered if any
 
 ## Validation
 
-*How did you validate that the construct and the search algorithm works?*
+We ran the algorithms known input and compared them to expected output. 
+Those tests can bee seen in test/tests.c and test/testSearch.c. 
+
+We also compared the results for randomly generated sequences and 
+patterns with the results from project 1.
 
 ## Running time
 
-*List experiments and results that show that both the construction algorithm and the search algorithm works in the expected running time. Add figures by embedding them here, as you learned how to do in project 1.*
+We ran the code on sequences of all A's for varying n and m. 
+Since the alphabet is constant parsing x should take O(n). Constructing the suffix array
```

Comment:
Yes, it is the re-mapping we refer to, since this is done in the parser. 
Since we are working with printable ascii, this will at most add O(128)=O(1)

Reply:
I think parsing should take linear time for any alphabet where you can access a letter in constant time, whether that alphabet is bounded by a constant or not. Is it the re-mapping you are referring to? If you are working with C strings, then they are sequences of `char` (which, although the standard doesn't require this, will always be bytes), so there can at most be 256 of those, so as long as you are working with `char` the alphabet has constant time. The same could be said about `int` and such, of course, but there we pretend that those can have size up to $O(\log n)$ bits, so we do not consider those constant. (In any case you wouldn't consider four billion a small constant, and that is what you would get with 32-bit integers, which is what `int` is on most platforms).